Add PFH in buck for FoA

lucylq opened the pull request:
Summary:
See:
- https://www.internalfb.com/intern/wiki/Privacy_Graph/Product_Asset_Graph/PFH_Asset_Linking/PFH_in_Buck_Targets/
- https://www.internalfb.com/intern/wiki/Privacy_Graph/Product_Feature_Hierarchy/

Seems to be a new requirement for FoA, where all buck targets owned by the app must be tagged with the product name. Codegen.bzl is included by third-parties and the libraries created 'belong' to the target app? So we add the feature name to the libraries constructed in codegen.

See failure: D58557891

Differential Revision: D58620058

dbort reviewed (approved):
Looks good! To fix the build failures, you could try popping `feature` in https://github.com/pytorch/executorch/blob/7c0f4c2b6f87369ee20544aada33dc55078c6bc6/shim/xplat/executorch/build/env_interface.bzl#L129

lucylq committed:
Add PFH in buck for FoA (#3985)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/3985

https://github.com/pytorch/executorch/pull/3985

See:
- https://www.internalfb.com/intern/wiki/Privacy_Graph/Product_Asset_Graph/PFH_Asset_Linking/PFH_in_Buck_Targets/
- https://www.internalfb.com/intern/wiki/Privacy_Graph/Product_Feature_Hierarchy/

Seems to be a new requirement for FoA, where all buck targets owned by the app must be tagged with the product name. Codegen.bzl is included by third-parties and the libraries created in codegen are built by and owned by the third-party. This is triggering the Product-Feature Hierarchy (PFH) feature requirement.

This diff adds the feature name to the libraries constructed in codegen.

See failure: D58557891

Reviewed By: dbort

Differential Revision: D58620058
